src/popup/pages/Notifications.vue

```diff
@@ -1,207 +1,66 @@
-} from '../utils';
 import NotificationItem from '../components/NotificationItem.vue';
-import Tabs from '../components/tabs/Tabs.vue';
-import Tab from '../components/tabs/Tab.vue';
-import { IS_MOBILE_DEVICE } from '../../lib/environment';
-
-const DIR_ALL = 'all';
-const DIR_WALLET = 'wallet';
-const DIR_SUPERHERO = 'superhero';
+import InfiniteScroll from '../components/InfiniteScroll.vue';
+import { useNotifications } from '../../composables/notifications';
+import { IS_EXTENSION } from '../../lib/environment';
 
 export default defineComponent({
   name: 'Notifications',
   components: {
+    InfiniteScroll,
     NotificationItem,
-    Tabs,
-    Tab,
   },
   setup(props, { root }) {
-    const direction = ref(DIR_ALL);
-
-    const notifications = computed<INotification[]>(
-      () => root.$store.state.notifications,
-    );
-
-    const notificationSettings = computed<INotificationSetting[]>(
-      () => root.$store.state.notificationSettings,
-    );
-
-    const notificationSettingsCheckedTypes = computed<string[]>(
-      () => notificationSettings.value.filter((s) => s.checked).map((s) => s.type),
-    );
-
-    const superheroNotifications = rxJsObservableToVueState(
-      root.$store.state.observables.notifications,
-      [],
-    ) as Ref<INotification[]>;
-
-    const filteredNotifications = computed<INotification[]>(
-      () => [...superheroNotifications.value, ...notifications.value]
-        .filter(
-          ({ type }) => (
-            direction.value === DIR_ALL
-            || (direction.value === DIR_SUPERHERO && type !== NOTIFICATION_TYPE_WALLET)
-            || (direction.value === DIR_WALLET && type === NOTIFICATION_TYPE_WALLET)
-          ),
-        )
-        .filter(
-          ({ status, type }) => (
-            status === NOTIFICATION_STATUS_READ
-            || notificationSettingsCheckedTypes.value.includes(type)
-          ),
-        )
-        .sort((a, b) => new Date(b.createdAt).getTime() - new Date(a.createdAt).getTime()),
-    );
-
-    async function modifyNotificationStatus(
-      notification: INotification,
-      status: NotificationStatus,
-    ) {
-      if (notification.type === NOTIFICATION_TYPE_WALLET) {
-        root.$store.commit('setNotificationsStatus', { createdAt: notification.createdAt, status });
-      } else {
-        await root.$store.dispatch('modifyNotification', [notification.id, status]);
-
-        const storedNotification = superheroNotifications.value
-          .find((n) => n.id === notification.id);
-
-        if (storedNotification) {
-          storedNotification.status = status;
-        }
+    const notificationSettings = computed(() => root.$store.state.notificationSettings);
+
+    const {
+      notificationsToShow,
+      canLoadMore,
+      loadMore,
+      markAsReadAll,
+    } = useNotifications(notificationSettings.value);
+
+    onMounted(() => {
+      loadMore();
+      if (IS_EXTENSION) {
+        root.$nextTick(() => {
```

Comment:
`nextTick` can be imported from the Composition Api to be typed properly.

Reply:
yes, and  please use await not callBack